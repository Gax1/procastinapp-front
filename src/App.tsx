import { useMemo } from "react";
import { Navigate, Route, Routes } from "react-router-dom";
import AuthLogin from "./components/AuthComponent/AuthLoginComponent";
import { Loading } from "./components/Loading/Loading";
import { Modal } from "./components/Modal/Modal";
import CreateTaskPage from "./pages/CreateTaskPage/CreateTaskPage";
import { DayTasksPage } from "./pages/DayTasksPage/DayTasksPage";
import { LoginPage } from "./pages/LoginPage/LoginPage";
import { NotFoundPage } from "./pages/NotFoundPage/NotFoundPage";
import { RegisterPage } from "./pages/RegisterPage/RegisterPage";
import { loginUserActionCreator } from "./store/features/usersSlice/usersSlice";
import { useAppDispatch, useAppSelector } from "./store/hooks/hooks";
import { RootState } from "./store/store";

function App() {
  const dispatch = useAppDispatch();
  const { isLoadding } = useAppSelector(
    (state: RootState) => state.ui.notification
  );

  const user = localStorage.getItem("user");
  useMemo(() => {
    if (user) {
      dispatch(loginUserActionCreator(JSON.parse(user)));
    }
  }, [user, dispatch]);

  return (
    <>
      <div className="main-container">
        {isLoadding && <Loading />}
        <Modal />
        <Routes>
          <Route path="/" element={<Navigate to={"/login"} />} />
          <Route path="/register" element={<RegisterPage />} />
          <Route path="/login" element={<LoginPage />} />
          <Route
            path="/my-day"
            element={<AuthLogin children={<DayTasksPage />} />}
          />
          <Route
            path="/my-day/create-task"
            element={<AuthLogin children={<CreateTaskPage />} />}
          />
          <Route path="*" element={<NotFoundPage />} />
        </Routes>
      </div>
    </>
  );
}

export default App;
