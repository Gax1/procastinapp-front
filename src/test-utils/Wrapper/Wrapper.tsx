import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { WrapperProps } from "../../interfaces/interfaces";
import { store } from "../../store/store";
import { theme } from "../../themes/theme";

export const Wrapper = ({ children }: WrapperProps) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <Provider store={store}>{children}</Provider>;
      </ThemeProvider>
    </BrowserRouter>
  );
};
