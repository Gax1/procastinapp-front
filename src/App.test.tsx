import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import App from "./App";
import { MockedWrapper, Wrapper } from "./test-utils/Wrapper/Wrapper";

describe("Given the app component", () => {
  describe("When rendered", () => {
    test("Then it should always match this snapshot", () => {
      const expectedApp = TestRenderer.create(
        <Wrapper>
          <App />
        </Wrapper>
      );

      expect(expectedApp).toMatchSnapshot();
    });
  });
  describe("When rendered with loading true", () => {
    test("Then it should show a modal", () => {
      jest.spyOn(Object.getPrototypeOf(window.localStorage), "getItem");
      Object.setPrototypeOf(window.localStorage.getItem, {
        user: { id: "test-id" },
      });
      render(
        <MockedWrapper>
          <App />
        </MockedWrapper>
      );

      const login = screen.getByTestId("loader-container");

      expect(login).toBeInTheDocument();
    });
  });
});
