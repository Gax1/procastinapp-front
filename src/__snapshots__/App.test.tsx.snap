// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the app component When rendered Then it should always match this snapshot 1`] = `
Array [
  <div
    className="main-container"
  />,
  ";",
]
`;
